ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh

```diff
@@ -24,6 +24,21 @@ function cleanup_ibmcloud_powervs() {
 
   echo "Cleaning up the Transit Gateways"
   RESOURCE_GROUP_ID=$(ic resource groups --output json | jq -r '.[] | select(.name == "'${resource_group}'").id')
+  for GW in $(ic tg gateways --output json | jq --arg resource_group "${RESOURCE_GROUP_ID}" --arg workspace_name "${WORKSPACE_NAME}" -r '.[] | select(.resource_group.id == $resource_group) | select(.name == $workspace_name) | "(.id)"')
+  do
+    VPC_CONN="${WORKSPACE_NAME}-vpc"
+    VPC_CONN_ID="$(ic tg connections "${GW}" 2>&1 | grep "${VPC_CONN}" | awk '{print $3}')"
+    if [ ! -z "${VPC_CONN_ID}" ]
+    then
+      echo "deleting VPC connection"
+      ic tg connection-delete "${GW}" "${CS}" --force || true
+      sleep 120
+      echo "Done Cleaning up GW VPC Connection"
+    else
+      echo "no vpc-conn needed"
+    fi
+  done
```

Comment:
```suggestion
  do
    VPC_CONN="${WORKSPACE_NAME}-vpc"
    VPC_CONN_ID="$(ic tg connections "${GW}" 2>&1 | grep "${VPC_CONN}" | awk '{print $3}')"
    if [ ! -z "${VPC_CONN_ID}" ]
    then
      echo "deleting VPC connection"
      ic tg connection-delete "${GW}" "${CS}" --force || true
      sleep 120
      echo "Done Cleaning up GW VPC Connection"
    else
      echo "no vpc-conn delete"
    fi
    break
  done
```

You have to delete every time. This comment is incorrect
Cleanup does cleanup commands.

Reply:
This one is resolved